drivers/storage/portworx/preflight.go

```diff
@@ -245,9 +278,31 @@ func (u *preFlightPortworx) ProcessPreFlightResults(recorder record.EventRecorde
 	}
 
 	if passed {
-		// Enable DMthin via misc args
-		u.cluster.Annotations[pxutil.AnnotationMiscArgs] = strings.TrimSpace(u.cluster.Annotations[pxutil.AnnotationMiscArgs] + " -T dmthin")
-		k8sutil.InfoEvent(recorder, u.cluster, util.PassPreFlight, "Enabling DMthin")
+		if !u.hardFail { // Enable DMthin via misc args if not enabled already
+			u.cluster.Annotations[pxutil.AnnotationMiscArgs] = strings.TrimSpace(u.cluster.Annotations[pxutil.AnnotationMiscArgs] + " -T dmthin")
+			// Only set if CloudStorage is nil
+			if u.cluster.Spec.CloudStorage == nil {
```

Comment:
- Let's add the metadata device in both cases where the user has passed `-T dmthin` or we added it for them.
- We need to handle the case where cloud storage is not nil. 
- We should not overwrite the metadata device if already provided by the user.
```
if u.cluster.Spec.CloudStorage == nil {
  u.cluster.Spec.CloudStorage = &corev1.CloudStorageSpec{}
}

if u.cluster.Spec.CloudStorage.SystemMdDeviceSpec != nil {
  u.cluster.Spec.CloudStorage.SystemMdDeviceSpec = &cmetadata
}
```